flink-runtime/src/main/java/org/apache/flink/runtime/state/changelog/ChangelogStateBackendHandle.java

```diff
@@ -44,28 +49,29 @@
  */
 @Internal
 public interface ChangelogStateBackendHandle extends KeyedStateHandle {
-    List<KeyedStateHandle> getMaterializedStateHandles();
+    Map<UUID, KeyedStateHandle> getMaterializedStateHandles();
```

Comment:
WDYT about using `StateHandleID` or `SharedStateRegistryKey` instead of `UUID` here and in other places?
That would be more flexible I think (allowing to change ID generation); and also more consistent with the current code - `IncrementalRemoteKeyedStateHandle` uses `StateHandleID` as keys in its map.

Reply:
I think `StateHandleID` is a good choice.